nylas/utils.py

```diff
@@ -7,8 +7,12 @@ def timestamp_from_dt(dt, epoch=datetime(1970, 1, 1)):
     Convert a datetime to a timestamp.
     https://stackoverflow.com/a/8778548/141395
     """
+    # For offset-aware datetime objects, convert them first before performing delta
+    if dt.tzinfo is not None and dt.utcoffset() is not None:
+        dt = dt.replace(tzinfo=None) - dt.utcoffset()
+
     delta = dt - epoch
-    # return delta.total_seconds()
+
     return delta.seconds + delta.days * 86400
 
```

Comment:
maybe a good idea to use timedelta from the datetime package for adding or subtracting time values from datetime. 

Also, datetime has a timestamp() method, so that could be just: return delta.timestamp()

Reply:
Thanks for pointing my attention to `timedelta` it helps clean the code a bit! Regarding `datetime.timestamp()` I initially used that as the solution, but unfortunately it's only available from Python 3.3 and up, and we still support Python 2.7 I believe so I had to scrap that idea.

Reply:
ok, this is good for py2, that timestamp() is in py3. datetime in python isn't the best package

Reply:
So I've heard :/